projects/js-packages/charts/src/components/leaderboard-chart/leaderboard-chart.tsx

```diff
@@ -0,0 +1,239 @@
+ *
+ * @param value - The delta value to format
+ * @return Formatted percentage string
+ */
+const defaultDeltaFormatter = ( value: number ): string => {
+	return formatMetricValue( value / 100, 'average', {
+		decimals: 0,
+		signDisplay: 'exceptZero',
+	} );
+};
+
+/**
+ * Progress bar component
+ *
+ * @param props           - Component props
+ * @param props.value     - Progress value as percentage (0-100)
+ * @param props.className - Additional CSS class name
+ * @param props.color     - Progress bar color
+ * @return JSX element representing the progress bar
+ */
+const ProgressBar: FC< {
+	value: number;
+	className?: string;
+	color?: string;
+} > = ( { value, className, color } ) => (
+	<div
+		className={ clsx( styles.progressBar, className ) }
+		style={ { '--progress-color': color } as React.CSSProperties }
+	>
+		<div
+			className={ styles.progressBarFill }
+			style={ { width: `${ Math.max( 0, Math.min( 100, value ) ) }%` } }
+		/>
+	</div>
+);
+
+/**
+ * LeaderboardChart component displays a ranked list of data with progress bars
+ * and optional comparison values.
+ *
+ * @param props                - Component props
+ * @param props.data           - Array of leaderboard entries to display
+ * @param props.withComparison - Whether to show comparison data
+ * @param props.primaryColor   - Primary color for current period bars
+ * @param props.secondaryColor - Secondary color for comparison period bars
+ * @param props.valueFormatter - Custom formatter for values
+ * @param props.deltaFormatter - Custom formatter for delta values
+ * @param props.loading        - Whether the chart is in loading state
+ * @param props.className      - Additional CSS class name
+ * @param props.style          - Custom styling for the chart container
+ * @return JSX element representing the leaderboard chart
+ */
+export const LeaderboardChart: FC< LeaderboardChartProps > = ( {
+	data,
+	withComparison = false,
+	primaryColor = '#3858E9',
+	secondaryColor = '#66BDFF',
+	valueFormatter = defaultValueFormatter,
+	deltaFormatter = defaultDeltaFormatter,
+	loading = false,
+	className,
+	style,
+} ) => {
+	const signColors = [ '#D63638', '#757575', '#008A20' ];
+
+	const chartStyle = {
+		'--primary-color': primaryColor,
+		'--secondary-color': secondaryColor,
+		...style,
+	} as React.CSSProperties;
+
+	// Handle empty or undefined data
+	if ( ! data || data.length === 0 ) {
+		return (
+			<div
+				className={ clsx( styles.leaderboardChart, loading && styles.loading, className ) }
+				style={ chartStyle }
+			>
+				<div className={ styles.emptyState }>{ loading ? 'Loading...' : 'No data available' }</div>
+			</div>
+		);
+	}
+
+	return (
+		<div
+			className={ clsx( styles.leaderboardChart, loading && styles.loading, className ) }
+			style={ chartStyle }
+		>
+			{ data.map( entry => {
+				const colorIndex = Math.sign( entry.delta ) + 1;
+				const deltaColor = signColors[ colorIndex ];
+
+				return (
+					<Fragment key={ entry.id }>
```

Comment:
Do we need this Fragment instance wrapping the div here? I think we can remove it.

Reply:
That makes sense, we can move the key attribute to the next level. Updated in [41dcbc5](https://github.com/Automattic/jetpack/pull/44299/commits/41dcbc5126cccc64d4e1f40b47b3cf48d68f2d54).